### PR TITLE
examples: add examples of using reth-provider and instantiating an RPC on top of the DB

### PR DESCRIPTION
**Pull request by gakonst**
As title, adds 2 examples:
* Showcases how to open a Reth DB in RO mode and talk to it via `reth-provider`. Pretty neat / powerful. cc @joshstevens19 for this to improve your abstractions' usage
* Showcases how to instantiate a very thin ETH JSON RPC server using the RpcModules builder. This currently feels still too verbose, cc @mattsse in case I'm doing something wrong.

cc @onbjerg I am not 100% sure if this is the best way to do examples/ in the current repository structure, opening this as-is, and we can either merge it or defer to you on best integration path

ref #3533

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
can we instead have a single examples/Cargo.toml and then all the examples are:

```toml
[[example]]
name = "<>"
path = "<>.rs"
```

ref: https://github.com/tokio-rs/tokio/tree/master/examples

File: examples/rpc-db/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    // 2. Setup blcokchain tree to be able to receive live notifs
+    // TODO: Make this easier to configure
+    let provider = {
+        let consensus = Arc::new(BeaconConsensus::new(spec.clone()));
+        let exec_factory = ExecutionFactory::new(spec.clone());
+
+        let externals = TreeExternals::new(db.clone(), consensus, exec_factory, spec.clone());
+        let tree_config = BlockchainTreeConfig::default();
+        let (canon_state_notification_sender, _receiver) =
+            tokio::sync::broadcast::channel(tree_config.max_reorg_depth() as usize * 2);
```
Comment (mattsse) [inline examples/rpc-db/src/main.rs:46]: this seems not necessary at all, we shouldn't be needing this here and instead should be able to use the db as provider,

perhaps there are a few things missing, but all of this should not be needed here

Comment (gakonst) [reply examples/rpc-db/src/main.rs:46]: yeah that's what i thought was just not sure how to set up the DB for the RPC Module

**Comment by gakonst** [comment]
> can we instead have a single examples/Cargo.toml and then all the examples are:

I guess yeah, altho it does require writing everything included. I'm fine with it.

**Pull request merged**